tmc-langs-util/src/main/java/fi/helsinki/cs/tmc/langs/util/Main.java

```diff
@@ -0,0 +1,161 @@
+import java.util.Map;
+
+public class Main {
+
+    final static String EXERCISE_PATH = "exercisePath";
+    final static String OUTPUT_PATH = "outputPath";
+    private static TaskExecutor executor = new TaskExecutorImpl();
+
+    public Main(TaskExecutor executor) {
+        Main.executor = executor;
+    }
+
+    public static void main(String[] args) {
+        if (args == null || args.length == 0) {
+            printHelp();
+        }
+        run(args);
+        System.exit(0);
+    }
+
+    private static void printHelp() {
+        System.out.println("\nUsage: Main <exercise path> <output path>\n"
+            + "\nOptions:\n"
+            + " --checkstyle <exercise path> <output path>\t\tRun checkstyle or similar plugin to project if applicable.\n"
+            + " --help\t\t\t\t\t\t\tDisplay help information.\n"
+            + " --preparesolution <exercise path>\t\t\tPrepare a presentable solution from the original.\n"
+            + " --preparestub <exercise path>\t\t\t\tPrepare a stub exercise from the original.\n"
+            + " --runtests <exercise path> <output path>\t\tRun the tests for the exercise.\n"
+            + " --scanexercise <exercise path> <output path>\t\tProduce an exercise description of an exercise directory.\n");
+        System.exit(0);
+    }
+
+    private static void run(String[] args) {
+        Map<String, Integer> commands = getCommands();
+        String command = args[0];
+        Integer pathsCount = commands.get(command);
+
+        if (pathsCount == null) {
+            printHelp();
+        } else if (!pathsCount.equals(args.length - 1)) {
+            System.out.println("ERROR: wrong argument count for " + command
+                + " expected " + pathsCount + " got " + (args.length - 1));
+            printHelp();
+        }
+
+        Map<String, Path> paths = parsePaths(args, pathsCount);
+
+        switch (command) {
+            case "--help":
+                printHelp();
+                break;
+            case "--checkstyle":
+                runCheckCodeStyle(paths);
+                break;
+            case "--scanexercise":
+                runScanExercise(paths);
+                break;
+            case "--runtests":
+                runTests(paths);
+                break;
+            case "--preparestub":
+                runPrepareStub(paths);
+                break;
+            case "--preparesolution":
+                runPrepareSolution(paths);
+                break;
+            default:
+                printHelp();
+                break;
+        }
+    }
+
+    private static Map<String, Integer> getCommands() {
+        //Command name and required argument count
+        Map<String, Integer> commands = new HashMap<>();
+        commands.put("--help", 0);
+        commands.put("--checkstyle", 2);
+        commands.put("--scanexercise", 2);
+        commands.put("--runtests", 2);
+        commands.put("--preparestub", 1);
+        commands.put("--preparesolution", 1);
```

Comment:
I think programs whose first argument has to be a command typically have that command not have the double dash in front of it. Some examples: git, svn, VirtualBox, virsh. (i.e. it's `git clone`, not `git --clone`)

In addition to the `help` command, I'd also support `-h` and `--help` at any position.